### PR TITLE
Check function arguments and parameters types

### PR DESCRIPTION
**Pull request by leewei05**
Resolves: #85

### CODE REVIEW

File: src/type_checker.cpp [hunk 1]
```diff
@@ -67,15 +68,18 @@ void TypeChecker::Visit(FuncDefNode& func_def) {
...
   } else {
+    std::vector<ExprType> param_types{};
+    for (auto& parameter : func_def.parameters) {
+      parameter->Accept(*this);
+      param_types.push_back(parameter->type);
+    }
+
     auto symbol = std::make_unique<SymbolEntry>(func_def.id);
     symbol->expr_type = func_def.return_type;
+    symbol->param_types = param_types;
```
Comment (Lai-YT) [inline src/type_checker.cpp:79]: Is there any particular reason preventing you from constructing the symbol beforehand and directly pushing the parameter types back into `symbol->param_types` during the visitation of parameters?

Here's the code that expresses my intention:

```cpp
    auto symbol = std::make_unique<SymbolEntry>(func_def.id);
    symbol->expr_type = func_def.return_type;
    for (auto& parameter : func_def.parameters) {
      parameter->Accept(*this);
      symbol->param_types.push_back(parameter->type);
    }
    env_.Add(std::move(symbol));
```

I understand that these approaches differ slightly in error handling. Are we considering this aspect?

File: src/type_checker.cpp [hunk 2]
```diff
@@ -229,9 +233,20 @@ void TypeChecker::Visit(FunCallExprNode& call_expr) {
...
+  auto func_def = env_.LookUp(id_expr->id);
+  auto& param_types = func_def->param_types;
+  auto& args = call_expr.args;
+  if (param_types.size() != args.size()) {
+    // TODO: argument size doesn't match
+  }
+
+  for (auto i = std::size_t{0}; i < args.size(); ++i) {
+    args[i]->Accept(*this);
+    if (args[i]->type != param_types[i]) {
```
Comment (Lai-YT) [inline src/type_checker.cpp:247]:
```suggestion
    args.at(i)->Accept(*this);
    if (args.at(i)->type != param_types.at(i)) {
```

**Review by Lai-YT — CHANGES_REQUESTED** [review]

File: src/type_checker.cpp (hunk 1 above)
Comment (leewei05) [reply src/type_checker.cpp:79]: This one is better!

**Review by Lai-YT — APPROVED** [review]
LGTM :rocket:

**Pull request merged**